filter.go

```diff
@@ -8,60 +8,162 @@ import (
 	manet "github.com/multiformats/go-multiaddr-net"
 )
 
+type filterEntry struct {
+	f      *net.IPNet
+	reject bool
```

Comment:
The reject/RejectByDefault naming plays logical mind tricks some times, e.g. negating a RejectByDefault clause is evil 😛  I'd be in favour of creating a `action` enum type with two values (Accept, Reject), and naming these fields `DefaultAction` and `action`.